### PR TITLE
difftest: support dump and rebuild by profile

### PR DESCRIPTION
**Pull request by klin02**
In acclerated and multi-core verification cases, we need to adjust or build Difftest's Hardware side seperately.

We use json to record Difftest apply and finish Info of DUT. Then difftest can rebuild for single or multi cores according to Profile.

### CODE REVIEW

**Comment by klin02** [comment]
I'm thinking is it appropriate to to generate SimTop in DifftestMain. If not, I can seperate it to a single file.

**Comment by klin02** [comment]
Json profile will dump to $(NOOP_HOME)/build/generated-src/difftest_profile.json

Example usage of rebuilding Difftest：
make difftest_verilog PROFILE=../build/generated-src/difftest_profile.json NUM_CORES=2 CONFIG=ZSE

File: src/test/scala/DifftestMain.scala [hunk 1]
```diff
@@ -54,6 +59,55 @@ class DifftestTop extends Module {
   DifftestModule.finish("demo")
 }
 
+// Generate simulation interface based on Profile describing the instantiated information of design
+class SimTop(ProfileName: String, numCores: Int) extends Module {
```
Comment (poemonsense) [inline src/test/scala/DifftestMain.scala:63]: `profileName` instead of `ProfileName`

**Review by poemonsense — APPROVED** [review]
Others look good

**Review by poemonsense — APPROVED** [review]

**Pull request merged**